packages/pyright-internal/src/tests/ipythonMode.test.ts

```diff
@@ -560,3 +560,48 @@ function findCommentByOffset(tokens: TextRangeCollection<Token>, offset: number)
 
     return comment;
 }
+
+test('unused expression at end is not error', async () => {
+    const code = `
+// @filename: test.py
+// @ipythonMode: true
+//// 4[|/*marker*/|]
+    `;
+
+    verifyAnalysisDiagnosticCount(code, 0);
+});
+
+test('unused expression is error if not at end of cell', async () => {
+    const code = `
+// @filename: test.py
+// @ipythonMode: true
+//// 4[|/*marker*/|]
+////
+//// x = 1
+    `;
+
+    verifyAnalysisDiagnosticCount(code, 1);
+});
+
+test('unused expression is error if within another statement', async () => {
+    const code = `
+// @filename: test.py
+// @ipythonMode: true
+//// if True:
+////     4[|/*marker*/|]
+    `;
+
+    verifyAnalysisDiagnosticCount(code, 1);
+});
+
+function verifyAnalysisDiagnosticCount(code: string, expectedCount: number) {
```

Comment:
would it be too much if we actually check the error kind rather than just count?